OverlayPlugin.Core/MemoryProcessors/Combatant/LineCombatant.cs

```diff
@@ -0,0 +1,421 @@
+                { typeof(Combatant).GetField(nameof(Combatant.BNpcNameID)),       0 },
+                { typeof(Combatant).GetField(nameof(Combatant.TransformationId)), 0 },
+                { typeof(Combatant).GetField(nameof(Combatant.WeaponId)),         0 },
+                { typeof(Combatant).GetField(nameof(Combatant.TargetID)),         0 },
+                { typeof(Combatant).GetField(nameof(Combatant.ModelStatus)),      0 },
+            });
+
+            private static readonly string[] IgnoreFieldNames = new string[] {
+                // "ID" is always printed
+                nameof(Combatant.ID),
+                // Exclude "Effects" due to object complexity
+                nameof(Combatant.Effects),
+                // Excluded due to not being useful
+                // TODO: Maybe this should just add any field flagged as NonSerialized, if additional fields are added?
+                nameof(Combatant.RawEffectiveDistance),
+                // Excluded due to being included in many, many other lines
+                nameof(Combatant.CurrentHP),
+            };
+
+            // Fields that should be written out for add or full list of changes
+            public static readonly FieldInfo[] AllFields = typeof(Combatant).GetFields()
+                .Where((field) => !IgnoreFieldNames.Contains(field.Name))
+                .ToArray();
+
+            private static object GetDefault(Type type)
+            {
+                if (type.IsValueType)
+                {
+                    return Activator.CreateInstance(type);
+                }
+                if (type == typeof(string))
+                {
+                    return string.Empty;
+                }
+                return null;
+            }
+
+            public static readonly ReadOnlyDictionary<Type, object> DefaultValues =
+                new ReadOnlyDictionary<Type, object>(
+                    AllFields.Select((fi) => fi.FieldType).Distinct().ToDictionary((t) => t, (t) => GetDefault(t)));
+        }
+
+        private class CombatantStateInfo
+        {
+            public DateTime lastUpdated;
+            public Combatant combatant;
+        }
+
+        private Func<string, DateTime, bool> logWriter;
+
+        private CancellationTokenSource cancellationToken;
+
+        public LineCombatant(TinyIoCContainer container)
+        {
+            logger = container.Resolve<ILogger>();
+            ffxiv = container.Resolve<FFXIVRepository>();
+            if (!ffxiv.IsFFXIVPluginPresent())
+                return;
+            combatantMemoryManager = container.Resolve<ICombatantMemory>();
+            container.Resolve<LineInCombat>().OnInCombatChanged += (sender, args) =>
+            {
+                if (args.InGameCombatChanged)
+                {
+                    // Clear the state map when leaving combat
+                    if (!args.InGameCombat)
+                    {
+                        WriteClearLine();
```

Comment:
This might be misleading - one one hand, it's a neat way of dealing with wipes and zone changes, since you'd implicitly be out of combat, but at the same time, "out of combat" doesn't necessarily equate to "there are no combatants". 

On further thought, I'm not sure what the best way to handle this is. You can certainly have pre-pull ability usages and the like, for which it would be beneficial to have *some* combatants data pre-pull, even if it's a heavily reduced tick rate like the previous implementation.